modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/BaseProcessDiscovery.java

```diff
@@ -468,14 +469,59 @@ public DiscoveredResourceDetails discoverResource(Configuration pluginConfig, Re
                 + "] as user [" + user + "]. Did you provide the correct credentials?");
         }
 
-        HostPort hostPort = new HostPort(false);
-        HostPort managementHostPort = new HostPort(false);
+        HostPort hostPort = new HostPort(true);
+        HostPort managementHostPort = new HostPort(true);
         managementHostPort.host = hostname;
         managementHostPort.port = port;
-        String key = createKeyForRemoteResource(hostname + ":" + port);
-        String name = buildDefaultResourceName(hostPort, managementHostPort, productType, null);
-        //FIXME this is inconsistent with how the version looks like when autodiscovered
-        String version = productInfo.getProductVersion();
+        hostPort.port = HostConfiguration.DEFAULT_NATIVE_PORT;
+        String key, name, version;
+        // Assume the server is local and follow a similar flow as if it was auto-discovered
+        ASConnection connection = new ASConnection(ASConnectionParams.createFrom(serverPluginConfig));
+        File homeDir = serverPluginConfig.getHomeDir();
+        if (homeDir == null) {
+            homeDir = getHomeDirFromConnection(connection);
+            serverPluginConfig.setHomeDir(homeDir);
+        }
+        if (serverPluginConfig.getBaseDir() == null) {
+            serverPluginConfig.setBaseDir(getBaseDirFromConnection(connection));
+        }
+        if (serverPluginConfig.getConfigDir() == null) {
+            serverPluginConfig.setConfigDir(getConfigDirFromConnection(connection));
+        }
+
+        File hostXmlFile = serverPluginConfig.getHostConfigFile();
+        if (hostXmlFile == null) {
+            hostXmlFile = getHostXmlFileFromConnection(connection);
+            serverPluginConfig.setHostConfigFile(hostXmlFile);
+        }
+        File logDir = serverPluginConfig.getLogDir();
+        if (logDir == null) {
+            logDir = getLogDirFromConnection(connection);
+            serverPluginConfig.setLogDir(logDir);
+        }
+        File logFile = getLogFile(logDir);
+        initLogEventSourcesConfigProp(logFile.getPath(), pluginConfig);
+
+        HostConfiguration hostConfig;
+        try {
+            hostConfig = loadHostConfiguration(hostXmlFile);
+            serverPluginConfig.setApiVersion(hostConfig.getDomainApiVersion());
+            pluginConfig.setSimpleValue("hostXmlFileName", hostXmlFile.getName());
+            key = createKeyForLocalResource(serverPluginConfig);
+            name = buildDefaultResourceName(hostPort, managementHostPort, productType, hostConfig.getHostName());
+            version = getVersion(homeDir, productType);
+            serverPluginConfig.setLocal(true);
+        } catch (Exception exception){
```

Comment:
I was going to reply the same, but actually none of these throw any checked exception. So only a RuntimeException can be thrown by these methods. And the offending method is loadHostConfiguration() which throws Exception (and no other type). I think the code would be nicer if we would change that method to throw something else.

Reply:
I'm not asking to change HostConfiguration class, which uses Exception instead of a list of Exceptions for convenience I guess. But in the loadHostConfiguration, we could catch that Exception and throw something more sane. 

loadHostConfiguration is protected, but it's not inherited by anything or used outside this class (at least my IDEA doesn't show any).

Reply:
That is, now it catches IOException from createKeyForLocalResource also.

Reply:
> That is, now it catches IOException from createKeyForLocalResource also.

Right, will reduce the work on that `try` to avoid that.